File: src/main/java/com/piveguyz/empickbackend/employment/recruitmentProcess/command/domain/enums/StepType.java
package com.piveguyz.empickbackend.employment.recruitmentProcess.command.domain.enums;

public enum StepType {
	DOCUMENT(0, "서류"),
	PRACTICAL(1, "실무"),
	INTERVIEW(2, "면접");

	private final int code;
	private final String label;

	StepType(int code, String label) {
		this.code = code;
		this.label = label;
	}

	public int getCode() {
		return code;
	}

	public String getLabel() {
		return label;
	}

	public static StepType fromCode(int code) {
		for (StepType type : StepType.values()) {
			if (type.code == code) return type;
		}
		throw new IllegalArgumentException("Invalid StepType code: " + code);
	}
}

File: src/constants/employment/stepType.js
export const stepTypeOptions = [
    { value: 'DOCUMENT', label: '서류전형' },
    { value: 'PRACTICAL', label: '실무테스트' },
    { value: 'INTERVIEW', label: '면접' }
]

export const getStepTypeLabel = (type) => {
    const map = {
        DOCUMENT: '서류전형',
        PRACTICAL: '실무테스트',
        INTERVIEW: '면접'
    }
    return map[type] || type
}
